meta/checkenumlock.sh

```diff
@@ -37,3 +37,7 @@ echo "Checking for possible enum values shift (current branch vs origin/master)
 ./checkheaders.pl -s ../inc/ temp/inc/
 
 rm -rf temp
+# clean up the git changes as well
+# workaround fix for git --work-tree=temp/ checkout ...
+# after checkout from other branch, data will be left in git
+git stash
```

Comment:
Looks like we should not use "git --work-tree=temp/ checkout origin/master inc", it has impact on the working directory. Maybe we cannot assumption the source folder is clean.

Reply:
yes, so this workaround will try to stash all the changes, that includes the intended changes. for example, if someone changed some files in the inc and experimental folder but not committed, then by this method, there sill be a chance to recover them.

Reply:
enhanced the approach